Source/EasyNetQ/Producer/DefaultRpc.cs

```diff
@@ -305,11 +304,7 @@ CancellationToken cancellationToken
 
         protected virtual void OnResponderFailure<TRequest, TResponse>(IMessage<TRequest> requestMessage, string exceptionMessage, Exception exception)
         {
-            // HACK: I think we can live with this, because it will run only on exception, 
-            // it tries to preserve the default serialization behavior, 
-            // being able to also deserialize POCO objects that has constructors with parameters
-            // this avoids to introduce a custom class wrapper that will change the message payload
-            var body = JsonConvert.DeserializeObject<TResponse>(typeof(TResponse) == typeof(string) ? "''" : "{}");
+            var body = typeof(TResponse) == typeof(string) ? (TResponse)(object)"''" : Activator.CreateInstance<TResponse>();
```

Comment:
I'm not sure that it is good replacement, because the comment is talking not about types with parameterless constructor 😞

Reply:
I have fixed it in separate PR.